### DISCUSSION TITLE
How to delay the popover with floating-ui

### DISCUSSION THREAD

**Discussion by pl1zahd** [floating-ui/floating-ui#1442] [Q&A] [answered]
Hello,

I'm implementing a simple tooltip element with floating-ui in JS and want to delay the showing of the tooltip.
I read this was possible with popper but I could not find a way in floating-ui.

Can you tell me the best solution, if there is one?

**Comment by atomiks** [comment, marked as answer]
Both Popper & Floating UI only position an element. What you're describing is outside their scope, but can be achieved with `setTimeout` when changing the visibility state of the popover.

**Comment by devsebastian** [comment]
hi @pl1zahd,
Aleast in the case of React, you could use the [@floating-ui/react-dom-interactions](https://floating-ui.com/docs/FloatingPortal#:~:text=React%20DOM%20Interactions), to customsie the delay before the popup. In the[ useHover() ](https://floating-ui.com/docs/useHover) hook we could add restMs property to define the delay before the popup like so:

```useHover(context, {restMs: 1000}) // one second delay```
